src/bun.js/bindings/KeyObject.cpp

```diff
@@ -0,0 +1,2385 @@
+                named_curve = values[1];
+                break;
+            case CryptoKeyEC::NamedCurve::P521:
+                named_curve = values[2];
+                break;
+            default:
+                ASSERT_NOT_REACHED();
+                named_curve = WTF::emptyString();
+            }
+
+            obj->putDirect(vm, JSC::PropertyName(JSC::Identifier::fromString(vm, "namedCurve"_s)), JSC::jsString(vm, named_curve), 0);
+            return JSC::JSValue::encode(obj);
+        }
+        case CryptoAlgorithmIdentifier::Ed25519: {
+            auto* obj = JSC::constructEmptyObject(lexicalGlobalObject, lexicalGlobalObject->objectPrototype(), 1);
+            auto& wrapped = key->wrapped();
+            const auto& okp = downcast<WebCore::CryptoKeyOKP>(wrapped);
+            obj->putDirect(vm, JSC::PropertyName(JSC::Identifier::fromString(vm, "namedCurve"_s)), JSC::jsString(vm, okp.namedCurveString()), 0);
+            return JSC::JSValue::encode(obj);
+        }
+        default:
+            return JSC::JSValue::encode(JSC::jsUndefined());
+        }
+    }
+    return JSC::JSValue::encode(JSC::jsUndefined());
+}
+
+JSC::EncodedJSValue KeyObject__generateKeyPairSync(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame)
+{
+    auto count = callFrame->argumentCount();
+    auto& vm = lexicalGlobalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    if (count < 1) {
+        JSC::throwTypeError(lexicalGlobalObject, scope, "generateKeyPairSync requires 1 arguments"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+
+    auto type = callFrame->argument(0);
+    if (type.isUndefinedOrNull() || type.isEmpty() || !type.isString()) {
+        JSC::throwTypeError(lexicalGlobalObject, scope, "type is expected to be a string"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+    auto type_str = type.toWTFString(lexicalGlobalObject);
+    RETURN_IF_EXCEPTION(scope, encodedJSValue());
+
+    Zig::GlobalObject* zigGlobalObject = reinterpret_cast<Zig::GlobalObject*>(lexicalGlobalObject);
+    auto* structure = zigGlobalObject->JSCryptoKeyStructure();
+    // TODO: rsa-pss
+    if (type_str == "rsa"_s) {
+        if (count == 1) {
+            JSC::throwTypeError(lexicalGlobalObject, scope, "options.modulusLength and options.publicExponentVector are required for rsa"_s);
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+        auto* options = jsDynamicCast<JSC::JSObject*>(callFrame->argument(1));
+        if (options == nullptr) {
+            JSC::throwTypeError(lexicalGlobalObject, scope, "options is expected to be a object"_s);
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+        auto modulusLengthJS = options->getIfPropertyExists(lexicalGlobalObject, PropertyName(Identifier::fromString(vm, "modulusLength"_s)));
+        if (modulusLengthJS.isUndefinedOrNull() || modulusLengthJS.isEmpty() || !modulusLengthJS.isNumber()) {
```

Comment:
Only need one check here

```suggestion
        if (!modulusLengthJS.isNumber()) {
```

Reply:
Fixed on https://github.com/oven-sh/bun/pull/5940/commits/59ebd2d62bf1da38db66be1cb3e00fcd2c56d302